apps/condo/domains/common/utils/codegeneration/new.generate.hooks.ts

```diff
@@ -0,0 +1,220 @@
+
+        }, [rowAction, initialValues, onComplete])
+    }
+
+    function useSoftDelete (onComplete?: IOnCompleteType<GQLObject>) {
+        const [rowAction] = useMutation(gql.UPDATE_OBJ_MUTATION)
+
+        return useCallback(async (obj: IUUIDObject) => {
+            const sender = getClientSideSenderInfo()
+            const variables = {
+                id: obj.id,
+                data: {
+                    dv: 1,
+                    sender,
+                    deletedAt: dayjs().toISOString(),
+                },
+            }
+
+            const { data, errors } = await rowAction({ variables })
+
+            if (data && data.obj) {
+                if (isFunction(onComplete)) {
+                    onComplete(data.obj)
+                }
+
+                return data.obj
+            }
+
+            if (errors) {
+                console.warn(errors)
+                throw errors
+            }
+
+            throw new Error('Unknown useSoftDelete action result!')
+        }, [rowAction, onComplete])
+    }
+
+    function useObjects (variables: QueryVariables, options?: QueryHookOptions<IUseObjectsQueryReturnType<GQLObject>, QueryVariables>) {
+        const intl = useIntl()
+        const AccessDeniedError = intl.formatMessage({ id: 'AccessError' })
+        const ServerError = intl.formatMessage({ id: 'ServerErrorPleaseTryAgainLater' })
+
+        const { data, error, loading, refetch, fetchMore } = useQuery<IUseObjectsQueryReturnType<GQLObject>, QueryVariables>(gql.GET_ALL_OBJS_WITH_COUNT_QUERY, {
+            variables,
+            notifyOnNetworkStatusChange: true,
+            ...options,
+        })
+
+        const objs: GQLObject[] = (data && data.objs) ? data.objs : []
+        const count = (data && data.meta) ? data.meta.count : null
+        const typedRefetch: IRefetchType<GQLObject, QueryVariables> = refetch
+        const typedFetchMore: IFetchMoreType<GQLObject, QueryVariables> = fetchMore
+
+        let readableError
+
+        if (error) {
+            readableError = String(error).includes('not have access') ? AccessDeniedError : ServerError
+            console.warn(error)
+        }
+
+        return {
+            loading,
+            objs,
+            count,
+            error: readableError,
+            refetch: typedRefetch,
+            fetchMore: typedFetchMore,
+        }
+    }
+
+    function useObject (variables: QueryVariables, options?: QueryHookOptions<IUseObjectsQueryReturnType<GQLObject>, QueryVariables>) {
+        const { objs, count, error, loading, refetch, fetchMore } = useObjects(variables, options)
+        if (count && count > 1) throw new Error('Wrong query condition! useObject hook must return single value!')
+        const obj = (objs && objs.length) ? objs[0] : null
+
+        return {
+            obj,
+            loading,
+            error,
+            refetch,
+            fetchMore,
+        }
+    }
+
+    return {
+        useNewCreate: useCreate,
```

Comment:
It has `new` prefix because of backward compatibility?

Reply:
For now I'm marking new hooks as `New` because of the need to see everything that was affected by it.
Now, then I'm finally done and all builds are passed. I can just remove new prefix, and everything will be working fine

Reply:
changed naming 👍